packages/openid4vc/src/openid4vc-verifier/router/federationEndpoint.ts

```diff
@@ -0,0 +1,133 @@
+  const rpRegistrationMetadataPayload: RPRegistrationMetadataPayload = {
+    id_token_signing_alg_values_supported: opts.idTokenSigningAlgValuesSupported,
+    request_object_signing_alg_values_supported: opts.requestObjectSigningAlgValuesSupported,
+    response_types_supported: opts.responseTypesSupported,
+    scopes_supported: opts.scopesSupported,
+    subject_types_supported: opts.subjectTypesSupported,
+    subject_syntax_types_supported: opts.subject_syntax_types_supported || ['did:web:', 'did:ion:'],
+    vp_formats: opts.vpFormatsSupported,
+    client_name: opts.clientName,
+    logo_uri: opts.logo_uri,
+    tos_uri: opts.tos_uri,
+    client_purpose: opts.clientPurpose,
+    client_id: opts.client_id,
+  }
+
+  const languageTagEnabledFieldsNamesMapping = new Map<string, string>()
+  languageTagEnabledFieldsNamesMapping.set('clientName', 'client_name')
+  languageTagEnabledFieldsNamesMapping.set('clientPurpose', 'client_purpose')
+
+  //   TODO: Do we need this?
+  const languageTaggedFields: Map<string, string> = LanguageTagUtils.getLanguageTaggedPropertiesMapped(
+    opts,
+    languageTagEnabledFieldsNamesMapping
+  )
+
+  languageTaggedFields.forEach((value: string, key: string) => {
+    const _key = key as keyof typeof rpRegistrationMetadataPayload
+    rpRegistrationMetadataPayload[_key] = value
+  })
+
+  return removeNullUndefined(rpRegistrationMetadataPayload)
+}
+
+export function configureFederationEndpoint(router: Router, config: OpenId4VcSiopFederationEndpointConfig) {
+  router.get(config.endpointPath, async (request: OpenId4VcVerificationRequest, response: Response, next) => {
+    const { agentContext, verifier } = getRequestContext(request)
+    const verifierService = agentContext.dependencyManager.resolve(OpenId4VcSiopVerifierService)
+    const verifierConfig = agentContext.dependencyManager.resolve(OpenId4VcVerifierModuleConfig)
+
+    try {
+      const { key } = await config.keyCallback(agentContext, {
+        verifierId: verifier.verifierId,
+      })
+
+      const relyingParty = await verifierService.getRelyingParty(agentContext, verifier.verifierId, {
+        clientId: verifierConfig.baseUrl,
+        clientIdScheme: 'entity_id',
+        authorizationResponseUrl: `${verifierConfig.baseUrl}/siop/${verifier.verifierId}/authorize`,
+      })
+
+      const verifierEntityId = `${verifierConfig.baseUrl}/${verifier.verifierId}`
+
+      const rpMetadata = createRPRegistrationMetadataPayload(relyingParty.createRequestOptions.clientMetadata)
+
+      // TODO: We also need to cache the entity configuration until it expires
+      const now = new Date()
+      // TODO: We also need to check if the x509 certificate is still valid until this expires
+      const expires = new Date(now.getTime() + 1000 * 60 * 60 * 24) // 1 day
+
+      const jwk = getJwkFromKey(key)
+      const alg = jwk.supportedSignatureAlgorithms[0]
+      const kid = 'key-1'
```

Comment:
can an federation put restrictions on which key types are used for signing the entity configurations?